sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/bitwiseExpressions.scala

```diff
@@ -204,3 +206,51 @@ case class BitwiseCount(child: Expression)
     case LongType => java.lang.Long.bitCount(input.asInstanceOf[Long])
   }
 }
+
+object BitwiseGet {
+  def calculate(target: Int, pos: Int): Int = {
+    val source = target.toBinaryString
+    if (pos < 0) {
+      throw new IllegalArgumentException(s"Invalid bit position: $pos less than zero")
+    } else if (source.length <= pos) {
+      throw new IllegalArgumentException(s"Invalid bit position: $pos exceeds the " +
+        "bit length of the target integer")
+    }
+    val subString = UTF8String.fromString(source).substringSQL(pos.asInstanceOf[Int], 1)
+    val intWrapper = new UTF8String.IntWrapper()
+    subString.toInt(intWrapper)
+    intWrapper.value
+  }
+}
+
+@ExpressionDescription(
+  usage = "_FUNC_(expr, pos) - Returns the value of the bit (0 or 1) at the specified position.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(11, 0);
+       1
+      > SELECT _FUNC_(11, 2);
+       0
+  """,
+  since = "3.2.0",
+  group = "bitwise_funcs")
+case class BitwiseGet(left: Expression, right: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  override def inputTypes: Seq[IntegerType] = Seq(IntegerType, IntegerType)
```

Comment:
`IntegralType` seems better choice as all the other bitwise functions use it.

Reply:
Impala give the definition `GETBIT(integer_type a, INT position)`.
snowflake give the description
```
integer_expr
This expression must evaluate to a data type that can be cast to INTEGER.
```
Because `BitwiseGet` uses the ImplicitCastInputTypes, so byte, short, int could be casted as Integral.

Reply:
What about `LongType`

Reply:
ImplicitCastInputTypes cannot cast `LongType` to `IntegerType`.
This expression must evaluate to a data type that can be cast to INTEGER.
Cast long to integer is not safe.

Reply:
That is why I propose not to use `ImplicitCastInputTypes`, but instead `ExpectsInputTypes` with `IntegralType` as the first input. So you could apply this also to Long type

Reply:
`checkConsistencyBetweenInterpretedAndCodegenAllowingException` seems to test the type cast. If `ImplicitCastInputTypes` could guarantee this function, we not need to add the test.
Most functions are not tested with `checkConsistencyBetweenInterpretedAndCodegenAllowingException`.

Reply:
The document of snowflake, impala uses integer type. teradata and yellowbrick uses long type.